### DISCUSSION TITLE
Creazione alarm_control_panel con chiamate REST

### DISCUSSION THREAD

**Discussion by ceceyax93** [indomus/forum#2373] [HUB: Home Assistant]
Ciao a tutti,
avrei bisogna di configurare un banale pannello di allarme su HomeAssistant e fare in modo che esistano i classici 3 scenari (home,away, disarm) che vadano a triggerare rispettivamente 3 rest_command già configurate e presenti nel configuration.yaml.
Successivamente mi piacerebbe che lo stato dello scenario attuale venga settato in base a una ulteriore chiamata REST che ritorna un JSON con un intero (1,2,3) in base allo scenario attivo.
Eventualmente, fatemi sapere se esiste una soluzione migliore (NodeRed?) per gestire una casistica del genere.

L'idea è di usare l'alarm_control_panel per poi bridgarlo su HomeKit.

Riuscireste ad aiutarmi con la scrittura del codice?
Ho provato a consultare la documentazione di HomeAssistant ma non riesco a trovare un esempio esplicativo che risponda alla mia esigenza.

Grazie mille.

**Comment by snakuzzo** [comment]
Dovresti usare questo -> https://www.home-assistant.io/integrations/alarm_control_panel.template/#configuration

**Reply by ceceyax93** [reply]
> Dovresti usare questo -> https://www.home-assistant.io/integrations/alarm_control_panel.template/#configuration

Si, ho provato a usarlo e a mettere il relativo "service" (le rest_command) per ogni scenario, ma quando lo aggiungo in UI vedo semplicemente il bottone "disarm" che tra l'altro non mi va neanche a triggerare il service configurato.

Premetto che sono abbastanza nabbo e sto ancora cercando di capire come funzionano entità, script ecc.

Grazie

**Reply by ceceyax93** [reply]
Questo è il codice che ho attualmente (ho mascherato le URL per privacy)

```# Allarme - Scenario ACCESO TUTTO
rest_command:
  allarme_accesotutto:
    url: 'https://XXXXX’'
    method: GET

# Allarme - Scenario NOTTE
  allarme_notte:
    url: 'https://XXXXX’'
    method: GET

# Allarme - Scenario SPENTO
  allarme_spento:
    url: 'https://XXXXX’
    method: GET

alarm_control_panel:
  - platform: template
    panels:
      safe_alarm_panel:
        unique_id: "allarmecasa"
        value_template: "{{ states('alarm_control_panel.real_alarm') }}"
        arm_away:
          service: rest_command.allarme_accesotutto
          target:
            entity_id: alarm_control_panel.real_alarm
        arm_home:
          service: rest_command.allarme_notte
          target:
            entity_id: alarm_control_panel.real_alarm
        disarm:
          - condition: state
            entity_id: device_tracker.paulus
            state: "home"
          - service: rest_command.allarme_spento
            target:
              entity_id: alarm_control_panel.real_alarm

```
